plugins/factory-plugin/src/devfile.ts

```diff
@@ -0,0 +1,44 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import * as theia from '@theia/plugin';
+import * as che from '@eclipse-che/plugin';
+
+const CREATE_WORKSPACE_COMMAND = {
+    id: 'factory-plugin.create-workspace',
+    label: 'Create Workspace',
+    tooltip: 'Create workspace from this DevFile'
+};
+
+export class Devfile {
+
+    constructor(private context: theia.PluginContext) {
+    }
+
+    init() {
+        this.context.subscriptions.push(theia.commands.registerCommand(
+            CREATE_WORKSPACE_COMMAND, uri => this.createWorkspace(uri))
+        );
+    }
+
+    async createWorkspace(uri: theia.Uri) {
```

Comment:
missing return type ?

Reply:
void

Reply:
It's a command handler. I suppose we can omit return type here. Isn't it?

Reply:
I don't know which tslint rules we're using but I was thinking it was mandatory

Reply:
It's not a problem to add return type after adding corresponding rule for tslint.